src/randomization.ts

```diff
@@ -62,10 +62,10 @@ type StringToUnknownMapping = {[key:string] : unknown};
  * @return true if the item may be appended to randomized without violating
  *         the constraints false otherwise.
  */
-function allowPushItem (
-    randomized: Array<StringToUnknownMapping>,
+function allowPushItem<TItem extends ItemType> (
```

Comment:
This is the magic that fixes your problem. It 'constrains' the generic to be (an extension of) a string-indexed object/dict/map/thing